src/opencv_example.cc

```diff
@@ -0,0 +1,128 @@
+  camera.set(cv::CAP_PROP_FRAME_WIDTH, width);
+  camera.set(cv::CAP_PROP_FRAME_HEIGHT, height);
+  camera.set(cv::CAP_PROP_FOURCC, camera_capture_format);
+  if (!camera.isOpened()) {
+    // If this is printed the camera might not be attached or it might not
+    // support the settings set above (width & height, capture API or format)
+    printf("Error: Could not open camera %d, verify the settings\n", camera_id);
+    return -1;
+  }
+
+  // Run the demo in an infinite loop
+  cv::Mat cv_image;
+  while (true) {
+    // Retrieve an image from the camera
+    camera.read(cv_image);
+
+    // Process the frame using the Plumerai People Detection library. This
+    // function is timed and framerate in FPS is reported. Note that the
+    // Plumerai People Detection library is not optimized for all systems,
+    // see the README for more details.
+    auto start_time = std::chrono::steady_clock::now();
+    auto predictions = ppd.process_frame(cv_image.data);
+    auto elapsed_time = std::chrono::steady_clock::now() - start_time;
+
+    // Report the time and framerate the above function took
+    auto time_ms =
+        std::chrono::duration<double, std::milli>(elapsed_time).count();
+    auto fps = static_cast<int>(1000 / time_ms);
+    printf("Processing took %.1f ms (%d fps)\n", time_ms, fps);
+
+    // Process the resulting bounding-boxes if there are any
+    for (auto &p : predictions) {
+      auto x_min =
+          std::max(0, std::min(static_cast<int>(p.x_min * width), width - 1));
+      auto y_min =
+          std::max(0, std::min(static_cast<int>(p.y_min * height), height - 1));
+      auto x_max =
+          std::max(0, std::min(static_cast<int>(p.x_max * width), width - 1));
+      auto y_max =
+          std::max(0, std::min(static_cast<int>(p.y_max * height), height - 1));
+
+      // Write the resulting coordinates to stdout
+      printf(
+          "Found box #%d with confidence %.2f at (x=%d, y=%d) - (x=%d, y=%d)\n",
+          p.id, p.confidence, x_min, y_min, x_max, y_max);
+
+      // Also draw the bounding-boxes on the video
+      auto colour = colours[p.id % num_colours];
+      for (int c = 0; c < 3; ++c) {
+        for (int x = x_min; x < x_max; x++) {
+          cv_image.data[(y_min * width + x) * 3 + c] = colour[c];
+          cv_image.data[(y_max * width + x) * 3 + c] = colour[c];
+        }
+        for (int y = y_min; y < y_max; y++) {
+          cv_image.data[(y * width + x_min) * 3 + c] = colour[c];
+          cv_image.data[(y * width + x_max) * 3 + c] = colour[c];
+        }
+      }
+
+      // Any user-processing code can be added here, e.g. writing the box
+      // coordinates into a file, sending them over the network, or modifying
+      // the image (e.g. person blurring for privacy).
+    }
+
+    // Finally display the webcam image with the resulting bounding-boxes
+    cv::imshow(window_text, cv_image);
+    cv::waitKey(30);  // can be changed to force a certain framerate
```

Comment:
It might be nice to exit on a keypress.
```suggestion
    if (cv::waitKey(30) & 0xFF == 27) {  // can be changed to force a certain framerate
      // Exit when ESC is pressed
      break;
    }
```
Or `if (waitKey() != -1)` to exit on _any_ key pressed.